File: wineserving/src/main/java/com/lightbend/java/wineserving/model/WineFactoryResolver.java
package com.lightbend.java.wineserving.model;

import com.lightbend.model.ModelFacroriesResolver;
import com.lightbend.model.ModelFactory;
import com.lightbend.model.Modeldescriptor;

import java.util.HashMap;
import java.util.Map;

// Factories resolver for Wine data
public class WineFactoryResolver implements ModelFacroriesResolver {

    private static final Map<Integer, ModelFactory> factories = new HashMap<Integer, ModelFactory>() {
        {
            put(Modeldescriptor.ModelDescriptor.ModelType.TENSORFLOW.getNumber(), WineTensorflowModelFactory.getInstance());
            put(Modeldescriptor.ModelDescriptor.ModelType.PMML.getNumber(), WinePMMLModelFactory.getInstance());
        }
    };

    @Override
    public ModelFactory getFactory(int type) {
        return factories.get(type);
    }
}

File: wineserving/src/main/scala/com/lightbend/wineserving/model/WineFactoryResolver.scala
package com.lightbend.wineserving.model

import com.lightbend.model.modeldescriptor.ModelDescriptor
import com.lightbend.modelServer.model.{ModelFactory, ModelFactoryResolver}

// Model factory resolver - requires specific factories
object WineFactoryResolver extends ModelFactoryResolver{

  private val factories = Map(ModelDescriptor.ModelType.PMML.value -> WinePMMLModel,
                              ModelDescriptor.ModelType.TENSORFLOW.value -> WineTensorFlowModel)

  override def getFactory(`type`: Int): Option[ModelFactory] = factories.get(`type`)
}
